### PR TITLE
Move Godot Physics 3D into a module; add dummy 3D physics server

### PR DESCRIPTION
**Pull request by rburing**
- *Production edit: 3D counterpart of https://github.com/godotengine/godot/pull/95261.*

If the module is enabled (default), 3D physics works as it did before this PR.

If the module is disabled and no other 3D physics server is registered (via a module or GDExtension), then Godot falls back to a dummy implementation which effectively disables 3D physics functionality (and a warning is printed).

To try it, compile with the SCons option `module_godot_physics_3d_enabled=no`.

This is useful for games that want to use a different 3D physics server, especially in web builds where binary size matters.

Linux binary size comparison (`scons target=template_release production=yes use_llvm=yes linker=lld`):

- Before (with Godot Physics 3D): 70.516 MiB (73941216 bytes)
- After (with only dummy 3D physics server): 69.603 MiB (72983072 bytes), nice.

So the binary size of the Godot Physics 3D module is 935.688 KiB (958144 bytes).

### CODE REVIEW

**Comment by Calinou** [comment]
Is this module automatically disabled with the `disable_3d=yes` SCons option?

File: modules/godot_physics_3d/config.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def can_build(env, platform):
+    return True
```
Comment (AThousandShips) [inline modules/godot_physics_3d/config.py:2]:
```suggestion
    return not env["disable_3d"]
```

**Comment by rburing** [comment]
@Calinou Thanks for the suggestion, the `disable_3d` option is taken into account now, and some `#ifndef`s that were made redundant were removed.

File: servers/physics_server_3d_dummy.h [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+	virtual void step(real_t p_step) override {}
+	virtual void sync() override {}
+	virtual void flush_queries() override {}
+	virtual void end_sync() override {}
+	virtual void finish() override {
+		memfree(body_state_dummy);
+		body_state_dummy = nullptr;
+		memfree(space_state_dummy);
+		space_state_dummy = nullptr;
+	}
```
Comment (rburing) [inline servers/physics_server_3d_dummy.h:435]: Even with this `finish()` I still get:

```
WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
     at: cleanup (core/object/object.cpp:2284)
Leaked instance: PhysicsDirectSpaceState3DDummy:7331643608
Leaked instance: PhysicsDirectBodyState3DDummy:7348420825
```

How should I do it properly?

**Review by Calinou — COMMENTED** [review]
Tested locally, it works as expected.

When compiling the editor with `module_godot_physics_3d_enabled=no`, the project settings dialog is correctly updated to remove the mention of the GodotPhysics server:

image

All 3D demo projects run when no physics server is enabled (with correct visuals), although you can't move, which is expected.

Some remarks (which also apply to the 2D counterpart of this PR, I assume):

- A warning is also printed on startup of an empty project with just a Node, both in the editor stdout (not visible in the Output panel as it's too early) and in the running project:

```
WARNING: No PhysicsServer3D implementation was found; 3D physics functionality will be disabled.
     at: initialize_physics (main/main.cpp:326)
```

- There should probably be a `Dummy` option you can select in the project settings for two reasons:
  - Silence the warning when you don't compile any physics engines. This is useful when compiling a binary with no physics engines to reduce the binary size of non-game applications, or games that write their own physics without using PhysicsServer.
  - Simulate what happens when physics are disabled at build-time, without needing to compile a separate binary.

**Comment by rburing** [comment]
Great idea @Calinou. I added the option, and now a warning only appears when the dummy server is used as a fallback:

> WARNING: Falling back to dummy PhysicsServer3D; 3D physics functionality will be disabled. If this is intended, set the physics/3d/physics_engine project setting to Dummy.

The warning is still not printed in the editor Output panel when the editor starts, but I guess that's fine.

**Review by Calinou — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Tested a build of a web `template_release` template, with the module:
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**